localstack-core/localstack/utils/diagnose.py

```diff
@@ -138,7 +138,13 @@ def traverse_file_tree(root: str) -> List[str]:
 
 
 def get_docker_image_details() -> Dict[str, str]:
-    return bootstrap.get_docker_image_details()
+    image = os.environ.get("IMAGE_NAME")
+    if not image:
+        try:
+            image = DOCKER_CLIENT.inspect_container(get_main_container_name())["Config"]["Image"]
+        except ContainerException:
+            return {}
+    return bootstrap.get_docker_image_details(image_name=image)
```

Comment:
The bootstrap method is mainly made for the CLI, it uses `get_docker_image_to_start` to determine the image to start. This is why `IMAGE_NAME` is important there, and why the function is only patched from -ext on the host, not inside the container.

I chose to alter the diagnostic module to avoid using a function clearly meant for being used from the CLI. Inside the running container we can determine the image from the current container, before starting the container, it cannot be done, so a different logic made sense.

I added some comments, and will check into a unit test. Unit tests have the same issue, though, as they are generally not running inside a container, so it is hard to test. So, I need to either mock a lot of the calls, or use a bootstrap test to test it using an image which has been retagged and altered in some way (so we can detect the hash difference).

Reply:
I added a unit test - it is not really pretty, as I mocked out the docker client quite a bit, but it asserts the behavior :)